### PR TITLE
New Xilinx Baremetal Assists file and Bug fixes and Improvements 

### PR DESCRIPTION
**Pull request by kedareswararao**
This pull request does the below
1) Added assist file for generating baremetal board support package config
2) Added assist file for generating xparameters.h file
3) Renamed the linker assist file
4) Generate the required conf files for yocto tool

### CODE REVIEW

File: assists/baremetalconfig_xlnx.py [hunk 1]
```diff
@@ -203,24 +205,55 @@ def is_compat(node, compat_string_to_test):
         return xlnx_generate_bm_config
     return ""
 
+def get_stdin(sdt, chosen_node, node_list):
+    prop_dict = Lopper.node_properties_as_dict(sdt.FDT, chosen_node.abs_path)
```
Comment (zeddii) [inline assists/baremetalconfig_xlnx.py:209]: Is the chosen node not being properly pulled into the SDT already ? If not, that's an issue that I need to address.

We shouldn't need to go to the FDT at all to process parts of the tree.

**Comment by zeddii** [comment]
Only one small comment from me. The assist cleanup and code generation look good.

If we can determine if there's a lopper bug based on my comment, or if we can change it to not read the FDT, I'll get the merge done immediately after that.

File: assists/baremetalconfig_xlnx.py (hunk 1 above)
Comment (kedareswararao) [reply assists/baremetalconfig_xlnx.py]: I have fixed it and force pushed the change. Please take a look into it.

**Comment by kedareswararao** [comment]
> Only one small comment from me. The assist cleanup and code generation look good.
> 
> If we can determine if there's a lopper bug based on my comment, or if we can change it to not read the FDT, I'll get the merge done immediately after that.

Thanks for the review @zeddii , I have fixed the bug and force pushed the change please review when you have some time and let me know your review comments 

Regards,
Kedar.

**Comment by zeddii** [comment]
Looks good to me. This is now merged.

**Pull request merged**